core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java

```diff
@@ -459,8 +459,21 @@ default void renameColumn(ConnectorSession session, ConnectorTableHandle tableHa
 
     /**
      * Drop the specified column
+     *
+     * @deprecated use {@link #dropColumn(ConnectorSession, ConnectorTableHandle, ColumnHandle, List)}
      */
+    @Deprecated
     default void dropColumn(ConnectorSession session, ConnectorTableHandle tableHandle, ColumnHandle column)
+    {
+        dropColumn(session, tableHandle, column, List.of());
```

Comment:
when introducing new method, the delegation should be the other way around

the engine should call the new method
the new method should have default implementation routing back to the old method

this ensures the old connectors remain source & binary compatible with the new version